user/security-in-qubes/firewall.md

```diff
@@ -242,270 +240,224 @@ When the qube `unstrusted` has started (after a first reboot), you can directly
 
-You can get this information from the Settings Window for the qube, or by running this command in each qube:
-`ifconfig | grep -i cast `
-Note the IP addresses you will need.
+You can get this information using various methods, but only the first one can be used for `sys-net` outside world IP:
+
+- by running this command in each qube: `ip -4 -br a | grep UP`
+- using `qvm-ls -n`
+- in the Qubes Manager window using the column IP
+- from the Settings Window for the qube
+
+Note the IP addresses you will need, they will be required in the next steps.
+
 > Note: The vifx.0 interface is the one used by qubes connected to this netvm so it is _not_ an outside world interface.
 
 **2. Route packets from the outside world to the FirewallVM**
 
-For the following example, we assume that the physical interface eth0 in sys-net has the IP address 192.168.x.y and that the IP address of sys-firewall is 10.137.1.z.
+For the following example, we assume that the physical interface ens6 in sys-net is on the local network 192.168.x.y with the IP 192.168.x.n, and that the IP address of sys-firewall is 10.137.1.z.
 
-In the sys-net VM's Terminal, code a natting firewall rule to route traffic on the outside interface for the service to the sys-firewall VM
+In the sys-net VM's Terminal, the first step is to to define an ntables chain that will receive DNAT rules to relay the network traffic on a given port to the qube NetVM, we recommend to define a new chain for each destination qube to ease rules management:
 
 ```
-iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 443 -d 192.168.x.y -j DNAT --to-destination 10.137.1.z
+nft add chain qubes custom-dnat-qubeDEST '{ type nat hook prerouting priority filter +1 ; policy accept; }'
 ```
 
-Code the appropriate new filtering firewall rule to allow new connections for the service
+> Note: the name `custom-dnat-qubeDST` is arbitrary
 
-```
-iptables -I FORWARD 2 -i eth0 -d 10.137.1.z -p tcp --dport 443 -m conntrack --ctstate NEW -j ACCEPT
-```
+> Note: while we use a DNAT chain for a single qube, it's totally possible to have a single DNAT chain for multiple qubes
 
-> If you want to expose the service on multiple interfaces, repeat the steps described in part 1 for each interface.
-> In Qubes R4, at the moment ([QubesOS/qubes-issues#3644](https://github.com/QubesOS/qubes-issues/issues/3644)), nftables is also used which imply that additional rules need to be set in a `qubes-firewall` nft table with a forward chain.
+Second step, code a natting firewall rule to route traffic on the outside interface for the service to the sys-firewall VM
 
-`nft add rule ip qubes-firewall forward meta iifname eth0 ip daddr 10.137.1.z tcp dport 443 ct state new counter accept`
+```
+nft add rule qubes custom-dnat-qubeDEST iifname == "ens6" ip saddr 192.168.x.y/24 tcp dport 443 ct state new,established,related counter dnat 10.137.1.z
```

Comment:
For a single specific interface it's better to use `iif` (it uses interface index internally, so it's slightly faster). Similar comment in few places below.

Reply:
Interesting difference https://serverfault.com/questions/985158/what-is-the-different-between-iifname-and-iif-in-nftables

(TLDR: iifname allows to add rules for interfaces that doesn't exist yet)